notebook/services/api/api.yaml

```diff
@@ -654,12 +639,18 @@ definitions:
       id:
         type: string
         format: uuid
-      notebook:
-        type: object
-        properties:
-          path:
-            type: string
-            description: path to notebook
+      path:
+        type: string
+        description: "The path to the session.  For console sessions, the path 
+         contains the cwd of the kernel and the name of the console session, 
+         which may be a uuid"
+      type:
+        type: string;
+        description: session type
+        enum:
+          - notebook
+          - file
+          - console
```

Comment:
From what I can tell in the current code, the only uniqueness is the tuple of `(session_id, path, kernel_id)` by virtue of the database.  So one could have multiple sessions for the same path, which I think is fine in a collaborative scenario.  The contents manager is expecting a full path to a file [here](https://github.com/jupyter/notebook/blob/master/notebook/services/sessions/sessionmanager.py#L89), so we'd have to make up a file name for a console, for example.
Given that these are *web* paths and not necessarily file paths, and don't have to reflect a file stored on disk, it may still make more sense then to just use `path` and let the "file name" be whatever one wants, such as a uuid.